File: firestore/app/src/main/java/com/google/example/firestore/SolutionDeletes.java
package com.google.example.firestore;

import androidx.annotation.NonNull;

import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.functions.FirebaseFunctions;
import com.google.firebase.functions.HttpsCallableReference;
import com.google.firebase.functions.HttpsCallableResult;

import java.util.HashMap;
import java.util.Map;

public class SolutionDeletes {

    // [START call_delete_function]
    /**
     * Call the 'recursiveDelete' callable function with a path to initiate
     * a server-side delete.
     */
    public void deleteAtPath(String path) {
        Map<String, Object> data = new HashMap<>();
        data.put("path", path);

        HttpsCallableReference deleteFn =
                FirebaseFunctions.getInstance().getHttpsCallable("recursiveDelete");
        deleteFn.call(data)
                .addOnSuccessListener(new OnSuccessListener<HttpsCallableResult>() {
                    @Override
                    public void onSuccess(HttpsCallableResult httpsCallableResult) {
                        // Delete Success
                        // ...
                    }
                })
                .addOnFailureListener(new OnFailureListener() {
                    @Override
                    public void onFailure(@NonNull Exception e) {
                        // Delete failed
                        // ...
                    }
                });
    }
    // [END call_delete_function]
}

File: firestore/app/src/main/java/com/google/example/firestore/kotlin/SolutionDeletes.kt
package com.google.example.firestore.kotlin

import com.google.firebase.functions.ktx.functions
import com.google.firebase.ktx.Firebase

class SolutionDeletes {

    // [START call_delete_function]
    /**
     * Call the 'recursiveDelete' callable function with a path to initiate
     * a server-side delete.
     */
    fun deleteAtPath(path: String) {
        val deleteFn = Firebase.functions.getHttpsCallable("recursiveDelete")
        deleteFn.call(hashMapOf("path" to path))
                .addOnSuccessListener {
                    // Delete Success
                    // ...
                }
                .addOnFailureListener {
                    // Delete Failed
                    // ...
                }
    }
    // [END call_delete_function]
}
